pkg/cluster/serviceprincipal.go

```diff
@@ -48,25 +44,21 @@ func (m *manager) clusterSPObjectID(ctx context.Context) error {
 	// wait.PollImmediateUntil. Doing this will not propagate the latest error
 	// to the user in case when wait exceeds the timeout
 	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
```

Comment:
Who do we need to wait 10m for service principal ID to be returned here, but not in `EnsureARMResourceGroupRoleAssignment()`?

Reply:
Speculating here, but maybe because this is production code?  Whereas `EnsureARMResourceGroupRoleAssignment` (the version that's not a no-op) only supports our development and CI environments where service principals are established, so maybe there's need for a retry loop there.